src/libstore/local-store.cc

```diff
@@ -451,7 +451,7 @@ AutoCloseFD LocalStore::openGCLock()
 {
     Path fnGCLock = stateDir + "/gc.lock";
     auto fdGCLock = open(fnGCLock.c_str(), O_RDWR | O_CREAT | O_CLOEXEC, 0600);
```

Comment:
An alternative could be to use `AutoCloseFD fdGCLock = open(...);` here instead, which will create a wrapper that will correctly compare the contained file descriptor to `-1` when implicitly converted to bool.
https://github.com/NixOS/nix/blob/d070d8b7460f412a657745698dba291c66792402/src/libutil/util.cc#L910-L913

This is used a couple of times around the codebase and maybe what was intended here.